pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -535,6 +542,51 @@ public PartitionedTopicMetadata deserialize(String key, byte[] content) throws E
         return metadataFuture;
     }
 
+    protected static PartitionedTopicMetadata fetchPartitionedTopicMetadataCheckAllowAutoCreation(
+            PulsarService pulsar, String path) {
+        try {
+            return fetchPartitionedTopicMetadataCheckAllowAutoCreationAsync(pulsar, path).get();
+        } catch (Exception e) {
+            if (e.getCause() instanceof RestException) {
+                throw (RestException) e;
+            }
+            throw new RestException(e);
+        }
+    }
+
+    protected static CompletableFuture<PartitionedTopicMetadata> fetchPartitionedTopicMetadataCheckAllowAutoCreationAsync(
+            PulsarService pulsar, String path) {
+        CompletableFuture<PartitionedTopicMetadata> metadataFuture = new CompletableFuture<>();
+        try {
+            boolean allowAutoTopicCreation = pulsar.getConfiguration().isAllowAutoTopicCreation();
+            String topicType = pulsar.getConfiguration().getAllowAutoTopicCreationType();
+            fetchPartitionedTopicMetadataAsync(pulsar, path).whenComplete((metadata, ex) -> {
+                if (ex != null) {
+                    metadataFuture.completeExceptionally(ex);
+                } else if (metadata.partitions == 0 && allowAutoTopicCreation &&
```

Comment:
Shall we also check if the non-partitioned topic exist or not? /cc @jiazhai

Reply:
Yes, we should. if non-partitioned topic exist, autoTopicCreation should not take effect.
There is also other places that need fix to avoid creating partitioned topic if non-partitioned topic exist.

@fxbing  would you please help fix this and add test for it?

Reply:
sure